secondHomeAdvanced.cpp

```diff
@@ -0,0 +1,97 @@
+
+#include "secondHomeAdvanced.hpp"
+
+/* 
+1. bool isExpressionValid (const std::string& expression)
+2. void linkLevelNodes(node* (или std::shared_ptr<node> - *КРАЙНЕ ПРЕДПОЧТИТЕЛЬНЕЕ ЕГО ИСПОЛЬЗОВАНИЕ!*)
+что такое node - см. задание
+
+1. std::vector<std::string> getUniqueWords (const std::string& text);
+Немного поучитесь перекидывать с одного контейнера данные в другой:
+2. std::vector<std::pair<std::string>,
+                       size_t> wordCounter (const std::string& text);
+*/
+
+bool isExpressionValid (const std::string& expression)   
+{
+    std::stack<char> stack;
+
+    for (auto ch : expression) {
```

Comment:
for basic types, it doesn't matter if you use copy (plain auto), but for objects it is critical, so please use const auto& whenever possible